test/e2e/e2e.go

```diff
@@ -130,12 +142,13 @@ var _ = Describe("validation", func() {
 		var metallbCRExisted bool
 
 		BeforeEach(func() {
-			metallb = &metallbv1beta1.MetalLB{}
-			err := loadMetalLBFromFile(metallb, consts.MetalLBCRFile)
-			Expect(err).ToNot(HaveOccurred())
-			metallb.SetNamespace(OperatorNameSpace)
+			metallb = getMetalLB()
+			if !SkipLoadingMetallbFromFile {
+				err := loadMetalLBFromFile(metallb, consts.MetalLBCRFile)
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
i think we still need to set the namespace after loading metallb from the file. currently the ns in the file is 'metallb-system', the same as set in the Makefile, so the CI is running without errors. once we use another ns for CI + load metallb from file the test will fail.

Reply:
done

Reply:
can we add that to the loadmetallb func instead of remembering to set it manually everytime?

Reply:
Moved loadMetallbFromFile to getMetalLB and returning error.
Expect is outside the functions

Reply:
so can we remove the other 2 setnamespaces in the file?
and another thing - does it matter where this is set?
https://github.com/metallb/metallb-operator/blob/16ca386fedbd9421f5effa516570afa2bf285880/test/e2e/e2e.go#L827
wouldn't loading from file afterwards override the namespace with what set in the file?

Reply:
We need the setnamespace if we don't use the resources loaded from file (as it will be used from cnf).
Added an if/else for more clarity.

Reply:
I meant these - are they still needed?
https://github.com/metallb/metallb-operator/blob/2535bfa3653dd21c7481e33684ee9da2760321ab/test/e2e/e2e.go#L340
https://github.com/metallb/metallb-operator/blob/2535bfa3653dd21c7481e33684ee9da2760321ab/test/e2e/e2e.go#L376

Reply:
Changed both.
The first one was not needed, the second one did not load from file, modified that.